### ISSUE TITLE
RawScrollbar does not apply track or border color

### ISSUE THREAD

**Issue by Piinks** [flutter/flutter#106917]
RawScrollbar.trackColor and RawScrollbar.trackBorderColor do not work.
Fix incoming.

**Comment by MrPiou94** [comment]
Hello @Piinks ,
This is my first contribution for Flutter, so I hope it will be sufficiently documented.
After a few tries here is a proposal for the track color..

trackVisibility: false, // Ignore colors well
![TrackFalse](https://user-images.githubusercontent.com/19947603/177339935-e354f0a6-baf6-4313-83e3-f5e4478fe7a3.PNG)

WithoutColor // Take default colors
![WithoutColor](https://user-images.githubusercontent.com/19947603/177340197-2591f20a-c918-48ed-95fa-0dabf20a544c.PNG)

WithColor // Respects the chosen colors
![WithColor](https://user-images.githubusercontent.com/19947603/177340376-b9b5b3dc-a29c-40ca-b1eb-28380b5a8c41.PNG)

To do this, I modified the following files;
flutter\packages\flutter\lib\src\widgets\scrollbar.dart

1) In the creation of RawScrollbar, give the default color
```dart
    this.trackColor,
    this.trackBorderColor,
```
```dart
    this.trackColor = const Color(0x08000000),
    this.trackBorderColor = const Color(0x1a000000),
```

2) Make colors not null because already initialized
```dart
  final Color? trackColor;
  final Color? trackBorderColor;
```
```dart
  final Color trackColor;
  final Color trackBorderColor;
```

3) Show colors in updateScrollbarPainter only if trackVisibility: true
```dart
      ..trackColor = _showTrack ? const Color(0x08000000) : const Color(0x00000000)
      ..trackBorderColor = _showTrack ? const Color(0x1a000000) : const Color(0x00000000)
```
```dart
      ..trackColor = _showTrack ? widget.trackColor : const Color(0x00000000)
      ..trackBorderColor = _showTrack ? widget.trackBorderColor : const Color(0x00000000)
```
I hope it will be useful for you, see you soon

**Comment by Piinks** [comment]
Hey @MrPiou94 thanks for the detailed review! I already had a fix ready for this, it is in #107177. I am working on a large refactoring in this file and came across the bug. 
I am happy to hear you are interested in contributing! If you have not already done so, please check out [the contributing guide](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md). I'll be happy to give you a review on any changes you submit. Thanks - and welcome!
